SignalMessaging/attachments/AttachmentApprovalViewController.swift

```diff
@@ -175,6 +228,90 @@ public class AttachmentApprovalViewController: OWSViewController {
         activityIndicatorView.autoCenterInSuperview()
         activityIndicatorView.startAnimating()
 
-        self.delegate?.didApproveAttachment()
+        self.delegate?.didApproveAttachment(attachment: attachment)
+    }
+}
+
+extension AttachmentApprovalViewController: UIScrollViewDelegate {
+
+    public func viewForZooming(in scrollView: UIScrollView) -> UIView? {
+        return mediaMessageView
+    }
+
+    fileprivate func updateMinZoomScaleForSize(_ size: CGSize) {
+        Logger.debug("\(logTag) in \(#function)")
+
+        // Ensure bounds have been computed
+        mediaMessageView.layoutIfNeeded()
+        guard mediaMessageView.bounds.width > 0, mediaMessageView.bounds.height > 0 else {
+            Logger.warn("\(logTag) bad bounds in \(#function)")
+            return
+        }
+
+        let widthScale = size.width / mediaMessageView.bounds.width
+        let heightScale = size.height / mediaMessageView.bounds.height
+        let minScale = min(widthScale, heightScale)
+        scrollView.maximumZoomScale = minScale * 5.0
+        scrollView.minimumZoomScale = minScale
+        scrollView.zoomScale = minScale
+    }
+
+    // Keep the media view centered within the scroll view as you zoom
+    public func scrollViewDidZoom(_ scrollView: UIScrollView) {
+        // The scroll view has zoomed, so you need to re-center the contents
+        let scrollViewSize = self.scrollViewVisibleSize
+
+        // First assume that mediaMessageView center coincides with the contents center
+        // This is correct when the mediaMessageView is bigger than scrollView due to zoom
+        var contentCenter = CGPoint(x: (scrollView.contentSize.width / 2), y: (scrollView.contentSize.height / 2))
+
+        let scrollViewCenter = self.scrollViewCenter
+
+        // if mediaMessageView is smaller than the scrollView visible size - fix the content center accordingly
+        if self.scrollView.contentSize.width < scrollViewSize.width {
+            contentCenter.x = scrollViewCenter.x
+        }
+
+        if self.scrollView.contentSize.height < scrollViewSize.height {
+            contentCenter.y = scrollViewCenter.y
+        }
+
+        self.mediaMessageView.center = contentCenter
+    }
+
+    // return the scroll view center
+    private var scrollViewCenter: CGPoint {
+        let size = scrollViewVisibleSize
+        return CGPoint(x: (size.width / 2), y: (size.height / 2))
+    }
+
+    // Return scrollview size without the area overlapping with tab and nav bar.
+    private var scrollViewVisibleSize: CGSize {
+        let contentInset = scrollView.contentInset
+        let scrollViewSize = scrollView.bounds.standardized.size
+        let width = scrollViewSize.width - contentInset.left - contentInset.right
```

Comment:
This is personal, but I always prefer to express this kind of thing like so:

```
scrollViewSize.width - (contentInset.left + contentInset.right)
```

Operator ordering can vary between languages and people tend to make mistakes around them, so I always try to be explicit and readable using parens.

Reply:
TIOLI obvs

Reply:
Agreed this is more legible, done!